MapboxDirections/MBVisualInstruction.swift

```diff
@@ -0,0 +1,82 @@
+import Foundation
+
+/**
+ :nodoc:
+ Encompasses all information necessary for creating a visual queue about a given `RouteStep`.
+ */
+@objc(MBVisualInstruction)
+public class VisualInstruction: NSObject, NSSecureCoding {
+    
+    /**
+     :nodoc:
+     Distance in meters from the beginning of the step at which the visual instruction should be visible.
+     */
+    public let distanceAlongStep: CLLocationDistance
```

Comment:
Consider factoring out `distanceAlongStep` from both VisualInstruction and SpokenInstruction into a new PositionedInstruction protocol. (Use `MBPositionedInstruction` as the Objective-C name if it’s necessary to expose this protocol to Objective-C.)